tests/core/io/test_xml_parser.h

```diff
@@ -230,6 +230,27 @@ TEST_CASE("[XMLParser] CDATA") {
 	CHECK_EQ(parser.get_node_type(), XMLParser::NodeType::NODE_ELEMENT_END);
 	CHECK_EQ(parser.get_node_name(), "a");
 }
+
+TEST_CASE("[XMLParser] Tag starting character(s)") {
+	SUBCASE("First character is a number") {
+		XMLParser parser;
+		const String input = "<1first></first>";
+		REQUIRE_EQ(parser.open_buffer(input.to_utf8_buffer()), OK);
+		REQUIRE_EQ(parser.read(), ERR_INVALID_DATA);
+	}
+	SUBCASE("First character is a punctuation") {
+		XMLParser parser;
+		const String input = "<.first></first>";
+		REQUIRE_EQ(parser.open_buffer(input.to_utf8_buffer()), OK);
+		REQUIRE_EQ(parser.read(), ERR_INVALID_DATA);
+	}
+	SUBCASE("First characters are 'xml") {
+		XMLParser parser;
+		const String input = "<xmlfirst></first>";
```

Comment:
At least have valid XML fragment in this case.
<xmlfirst></xmlfirst>
What you have is invalid xml and wouldn't expect it parse correctly.

Reply:
Are you saying that I should write it as <xmlfirst></xmlfirst>? 

Otherwise, I'm not sure what you are referring to specifically, but in that subcase I'm testing if an error gets thrown due to the fact that an xml tag cannot start with "xml".

Reply:
Yes that, is what I was implying.  We still one it to be valid xml syntax.

Reply:
Got it, I edited it in the last commit, thanks